students/Kristy_Martini/lesson09/mailroom.py

```diff
@@ -0,0 +1,149 @@
+import operator
+import os
+
+class Report:
+    def __init__(self):
+        """ Report class holds the list of all donors and information associated with them."""
+        self.donors = dict()
+        Kristy = Donor("Kristy Martini", 0, 3, 3000)
+        Mikey = Donor("Mike Martini", 0, 7, 3424834)
+        Cathy = Donor("Cathy Martini", 0, 7, 63833)
+        Bill = Donor("Bill Martini", 0, 10, 60000)
+        Nick = Donor("Nick Martini", 0, 50, 47484949)
+
+        initial_donors = [Kristy, Mikey, Nick, Bill, Cathy]
+        [self.add_donor(donor) for donor in initial_donors]
+
+    def add_donor(self, Donor):
+        """ Add new donor to report"""
+        self.donors[Donor.name] = Donor
+    
+    def sort_donors(self, new_report=None):
+        """ Sorts donors by total gift value given"""
+        if new_report is None:
+            new_report = self
+        new_report.unsorted_list = list(new_report.donors.values())
+        new_report.sorted_list = sorted(new_report.unsorted_list, key=operator.attrgetter('total_gift_value'), reverse=True)
+        new_report.sorted_dict = dict()
+        for donor in new_report.sorted_list:
+            new_report.sorted_dict[donor.name] = donor
+
+    def create_report(self, new_report=None):
+        """ Display donor report to the user"""
+        if new_report is None:
+            new_report = Report()
+        new_report.sort_donors(new_report)
+        report_lines = []
+        report_lines.append("Donor Name          | Total Given   | Num Gifts | Average Gift")
+        report_lines.append("--------------------------------------------------------------")
+        for value in new_report.sorted_dict.values():
+            new_line = '{0:21}'.format(value.name) + "$" + '{0:14}'.format(value.total_gift_value) + '{0:12}'.format(value.num_gifts) + " $" + '{0:12}'.format(value.average_gift)
+            report_lines.append(new_line)
+        report_lines.append("\n")
+        return report_lines
+
+    def display_report(self, new_report):
+        report_lines = new_report.create_report(new_report)
+        for line in report_lines:
+            terminal_print(line)
+        prompt_user(new_report)
```

Comment:
same here - `self.create_report` is how you would use your class method.

also, prompt_user should be outside of this class.